flocker/node/script.py

```diff
@@ -304,6 +264,17 @@ def flocker_dataset_agent_main():
     loopback block device backend.  Later it will be capable of starting a
     dataset agent using any of the support dataset backends.
     """
+    # XXX This should use dynamic dispatch in the deployer_factory
+    # There should be only AgentScript, not ZFSAgentScript, and it should
+    # do the right thing for the configured backend. FLOC-1791.
+
+    options = DatasetAgentOptions()
+
+    return FlockerScriptRunner(
```

Comment:
Sure, that is the entry point, but what used that entry point? I think nothing. I think that previously the only tests which used one of these agent entry points were the acceptance tests which used the ZFS. Now the acceptance tests still get a ZFS agent, but by going to the via the `flocker-dataset-agent` entry point.

The second test is just for `--version` and `--help` so nothing changed.